frontend/test/cypress/integration/product-list/filters.spec.ts

```diff
@@ -0,0 +1,167 @@
+const TEST_MIN_PRICE_INVALID = 100;
+const TEST_MIN_PRICE = 10;
+const TEST_MAX_PRICE = 50;
+
+describe('product list filters', () => {
+   const user = cy;
+   beforeEach(() => {
+      user.visit('/store/parts');
+   });
+
+   function getVirtualListClickOptions(): Partial<Cypress.ClickOptions> {
+      return {
+         // Cypress won't click the button because it detects that there's a parent
+         // element with the pointer-events: none style.
+         // Since this it's expected behavior for a virtual list while scrolling (
+         // https://github.com/bvaughn/react-window/issues/128#issuecomment-460163077 ),
+         // we need to force the click.
+         force: true,
+      };
+   }
+
+   function getVirtualListTypeOptions(): Partial<Cypress.TypeOptions> {
+      return {
+         // Cypress won't click the button because it detects that there's a parent
+         // element with the pointer-events: none style.
+         // Since this it's expected behavior for a virtual list while scrolling (
+         // https://github.com/bvaughn/react-window/issues/128#issuecomment-460163077 ),
+         // we need to force the click.
+         force: true,
+      };
+   }
+
+   function waitForSearchCompletion() {
+      user
+         .findByRole('region', { name: /products/i })
+         .as('section')
+         .should('have.attr', 'data-search-state', 'search');
+      return user
+         .get('@section')
+         .should('have.attr', 'data-search-state', 'idle');
+   }
+
+   it('should help user filter', () => {
+      user
+         .findByRole('button', { name: /expand item type/i, expanded: false })
```

Comment:
Out of curiosity, why did we decide to go with text instead of selectors to find elements? I'm thinking that the text is more likely to change than the selector as we develop/maintain this.

Reply:
I've iterated over this a few times (initially I was using `data-testid` everywhere), and I settled on the approach described [here](https://testing-library.com/docs/queries/about#priority) because to me it seems the best tradeoff. This approach tries to mimic how users interact with the page (and it does so in a way that takes into account also accessibility). 
Although I agree with you that text is more likely to change, it's probably better to update tests accordingly (I think this makes sure that we are intending to change expectation over the copy, and we're not breaking the semantic and accessibility by mistake).
The `name` regex tries to get an elements from their [accessible name](https://www.w3.org/TR/accname-1.1/), so this should prevent us from breaking accessibility.

Let me also say, that we probably need to do more work to make sure that the page is completely accessible, since I haven't performed any test with the screen reader yet, but I think this is a good start.